routemaster/jobs/batch.rb

```diff
@@ -35,21 +36,15 @@ def call(uid)
           batch.delete
         rescue Services::Deliver::CantDeliver => e
           _log_exception(e)
-          attempts = batch.fail
-          raise Models::Queue::Retry, _backoff(attempts)
+          raise Models::Queue::Retry, _backoff(batch)
         end
         self
       end
 
       private
       
-      def _backoff(attempts)
-        backoff = 1_000 * 2 ** [attempts-1, _backoff_limit].min
-        backoff + rand(backoff)
-      end
-
-      def _backoff_limit
-        @@_backoff_limit ||= Integer(ENV.fetch('ROUTEMASTER_BACKOFF_LIMIT'))
+      def _backoff(batch)
+        Services::Throttle.new(batch: batch).retry_backoff
```

Comment:
Not sure there's value in keeping the legacy system — per-batch backoff was a hack; I think it's more likely for a subscriber to be up/down globally than to be up/down for a given batch.
Up to you whether to support it (... but we'd need full test coverage for both)

Noted on the "hot" objects — that's cool. I think my implicit point was that it feels a little smelly to re-instantiate the throttle here, when we could keep communication with the throttle service entirely within the deliver service (= lower coupling).